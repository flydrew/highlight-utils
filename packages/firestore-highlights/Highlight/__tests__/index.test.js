const Highlight = require("../");

describe("Highlight", () => {
  let Doc;

  beforeEach(() => {
    Doc = {
      ref: { id: 1 },
      data: () => ({
        body: "Hello world",
        location: "123",
        hash: 12345,
        documentSentiment: {
          magnitude: 0.800000011920929,
          score: -0.4000000059604645
        },
        languageAnalysis: {
          entities: [
            { mentions: [{ type: "COMMON" }], name: "trip", type: "EVENT" },
            {
              name: "Cairo, Eqypt",
              type: "LOCATION",
              metadata: {
                wikipedia_url: "https://en.wikipedia.org/wiki/Cairo",
                mid: "/m/01w2v"
              },
              mentions: [{ type: "PROPER" }]
            }
          ]
        }
      })
    };
  });

  it("returns subset of Highlight attributes", () => {
    expect(Highlight.attrs(Doc)).toMatchSnapshot();
  });

  it("creates categories index", () => {
    const categories = [{ name: "Foo Bar" }, { name: "Alpha beta" }];
    expect(Highlight.indexCategories(categories)).toMatchSnapshot();
  });

  it("creates entities index", () => {
    expect(
      Highlight.indexEntities(Highlight.properEntities(Doc.data()))
    ).toMatchSnapshot();
  });
});
