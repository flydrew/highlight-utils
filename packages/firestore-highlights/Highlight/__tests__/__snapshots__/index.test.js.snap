// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlight creates categories index 1`] = `
Object {
  "alphaBeta": true,
  "fooBar": true,
}
`;

exports[`Highlight creates entities index 1`] = `
Object {
  "british": true,
  "cairoEqypt": true,
}
`;

exports[`Highlight returns subset of Highlight attributes 1`] = `
Object {
  "body": "Hello world",
  "entities": Array [
    Object {
      "metadata": Object {},
      "name": "Cairo, Eqypt",
      "type": "LOCATION",
    },
    Object {
      "metadata": Object {
        "wikipedia_url": "https://en.wikipedia.org/wiki/United_Kingdom",
      },
      "name": "British",
      "type": "LOCATION",
    },
  ],
  "id": 1,
  "location": "123",
  "sentiment": Object {
    "magnitude": 0.800000011920929,
    "score": -0.4000000059604645,
  },
}
`;
